src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionSender.cs

```diff
@@ -100,15 +164,105 @@ protected void OnTransmissionSent(TransmissionProcessedEventArgs args)
+        {
+            if (!this.applyThrottle)
+            {
+                return numEvents;
+            }
+
+            var throttleWindowId = (long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalMilliseconds / this.ThrottleWindow);
+            var isInThrottleWindow = throttleWindowId == currentThrottleWindowId;
+            if (isInThrottleWindow && currentItemsCount < this.ThrottleLimit)
+            {
+                var numAccepted = Math.Min(numEvents, this.ThrottleLimit - this.currentItemsCount);
+                if (!peek)
+                {
+                    this.currentItemsCount += numAccepted;
                 }
 
-                this.OnTransmissionSent(new TransmissionProcessedEventArgs(transmission, exception, responseContent));
+                return numAccepted;
+            }
+            else if (!isInThrottleWindow)
+            {
+                var numAccepted = Math.Min(numEvents, this.ThrottleLimit);
+                this.currentThrottleWindowId = throttleWindowId;
+                this.currentItemsCount = numAccepted;
+                return numAccepted;
+            }
+
+            return 0;
+        }
+
+        private Transmission Throttle(Transmission transmission)
+        {
+            if (!this.ApplyThrottle)
+            {
+                return transmission;
             }
+
+            int attemptedItemsCount = -1;
+            int acceptedItemsCount = -1;
+            var transmissions = transmission.Split((transmissionLength) => {
+                attemptedItemsCount = transmissionLength;
+                acceptedItemsCount = this.IsTransmissionSendable(transmissionLength);
+                return acceptedItemsCount;
+            });
+
+            Transmission acceptedTransmission = transmissions.Item1;
+            Transmission rejectedTransmission = transmissions.Item2;
+            
+            // Send rejected payload back for retry
+            if (rejectedTransmission != null)
+            {
+                TelemetryChannelEventSource.Log.TransmissionThrottledWarning(this.ThrottleLimit, attemptedItemsCount, acceptedItemsCount);
+                SendTransmissionThrottleRejection(rejectedTransmission);
+            }
+            
+            return acceptedTransmission;
+        }
+
+        private void SendTransmissionThrottleRejection(Transmission rejectedTransmission)
+        {
+            var exception = new WebException(
+                "Transmission was split by local throttling policy",
+                null,
+                System.Net.WebExceptionStatus.Success,
+                null
+            );
+            this.OnTransmissionSent(new TransmissionProcessedEventArgs(rejectedTransmission, exception, new HttpWebResponseWrapper()
+            {
+                StatusCode = ResponseStatusCodes.ResponseCodeTooManyRequests,
+                StatusDescription = "Internally Throttled",
+                RetryAfterHeader = null
```

Comment:
The policy that handles throttling would respect that rather than use exponential backoff. Is that better?